### PR TITLE
[App] Add users last seen t&c's version update to login flow

### PR DESCRIPTION
**Pull request by rners01**
project - https://www.notion.so/saladtech/Terms-Conditions-Update-Flow-565e181573ff4068bc7aa5881a798534
task - https://www.notion.so/saladtech/App-Add-login-register-update-users-last-seen-terms-e90af1576e874adb828e691200beee6a

### CODE REVIEW

File: packages/web-app/src/modules/terms-and-conditions/TermsAndConditionsStore.ts [hunk 1]
```diff
@@ -14,20 +14,26 @@ export class TermsAndConditionsStore {
...
+    private readonly profile: ProfileStore,
     ) {}
 
   @action.bound
   public submitTermsAndConditions = flow(function* (this: TermsAndConditionsStore) {
     try {
+      const pendingTermsVersion = this.profile.currentProfile?.pendingTermsVersion
+
+      if(!pendingTermsVersion) {
+        return
```
Comment (Maks19) [inline packages/web-app/src/modules/terms-and-conditions/TermsAndConditionsStore.ts:26]: Could we log what went wrong here?
As I understand `submitTermsAndConditions` will be triggered after button click so if `pendingTermsVersion` is false user won't see anything and we won't know what happened

File: packages/web-app/src/Store.ts [hunk 2]
```diff
@@ -189,6 +189,7 @@ export class RootStore {
         this.refresh.refreshData(),
         this.profile.loadPayPalId(),
         this.saladCard.loadSaladCard(),
+        this.termsAndConditions.submitTermsAndConditions(),
```
Comment (scottcinatl) [inline packages/web-app/src/Store.ts:192]: What is the expected experience on an error here? It seems that in the `submitTermsAndConditions` call that we are not throwing any error if one occurs. Is the user supposed to be able to continue if they aren't able to accept the new terms and conditions?

File: packages/web-app/src/modules/terms-and-conditions/TermsAndConditionsStore.ts (hunk 1 above)
Comment (rners01) [reply packages/web-app/src/modules/terms-and-conditions/TermsAndConditionsStore.ts:26]: no, this is for other purposes(Im reusing this logic for login/register flow, there could be no ```pendingTermsVersion``` so I don't want to send request)
if ```pendingTermsVersion``` is ```null```, then user wont even see this whole block with update T&Cs

File: packages/web-app/src/Store.ts (hunk 2 above)
Comment (rners01) [reply packages/web-app/src/Store.ts:192]: good question, with existing logic
if user will not be able to update T&Cs he will just keep seeing notification in the widget to update them and update T&Cs block in the account page in web-app

Comment (rners01) [reply packages/web-app/src/Store.ts:192]: I'll double check that with Ivan, to be sure that such behaviour satisfies is acceptable

Comment (rners01) [reply packages/web-app/src/Store.ts:192]: yeah, looks like Ivan will be back after holidays
What do you think then @scottcinatl ?

Comment (rners01) [reply packages/web-app/src/Store.ts:192]: checked with Ivan, he seems ok with existing solution
image

File: packages/web-app/src/modules/terms-and-conditions/TermsAndConditionsStore.ts (hunk 1 above)
Comment (rners01) [reply packages/web-app/src/modules/terms-and-conditions/TermsAndConditionsStore.ts]: @Maks19 updated the solution to drop additional check

**Review by scottcinatl — CHANGES_REQUESTED** [review]

File: packages/web-app/src/Store.ts [hunk 3]
```diff
@@ -192,6 +192,9 @@ export class RootStore {
         this.helpScout.login(profile.username, profile.email),
         Promise.race([this.saladBowl.login(), delay(10000)]),
       ])
+      if(profile.pendingTermsVersion) {
```
[the remark this answers is not in the document]
Comment (rners01) [reply packages/web-app/src/Store.ts:195]: its all possible, but I wanted to keep `submitTermsAndConditions` clean and simple, as we'r reusing it in couple places

**Review by scottcinatl — DISMISSED** [review]

File: packages/web-app/src/Store.ts (hunk 3 above)
Comment (scottcinatl) [inline packages/web-app/src/Store.ts:195]: I noticed this wasn't automatically formatted by prettier. There should be a space here, when you save. So make sure we are opening the project in the right folder so we get the correct vscode tooling. For instance, I think this project has to be opened in the `salad-applications`  folder. But when jumping around to each project let's just make sure we are opening it from the correct folder.

Comment (rners01) [reply packages/web-app/src/Store.ts:195]: Its not working, where do you see such config, I can't find it

Comment (rners01) [reply packages/web-app/src/Store.ts:196]: but anyway fixed,
we should add some eslint config file for it

**Pull request merged**